### PR TITLE
[Merged by Bors] - Add From<f32> for JsValue

### LINKED ISSUE #1989: Implement From<f32> for JsValue
I've noticed that the From trait for f32, i128, i16, i8, u128, u16, and u8 are not implemented for JsValue. This is my first time contributing directly to boa, and so I'm just fixing something small that I think is a problem. This is all in preparation for https://github.com/boa-dev/boa/issues/1975

### PR DESCRIPTION
**Pull request by lastmjs**
This Pull Request fixes/closes #1989.

It changes the following:

- Implements From<f32> for JsValue

### CODE REVIEW

File: boa_engine/src/value/conversions.rs [hunk 1]
```diff
@@ -54,6 +54,18 @@ impl From<f64> for JsValue {
...
 
+impl From<f32> for JsValue {
+    #[allow(clippy::float_cmp)]
+    #[inline]
+    fn from(value: f32) -> Self {
+        // if value as i32 as f32 == value {
+        //     Self::Integer(value as i32)
+        // } else {
+        Self::Rational(value.into())
+        // }
```
Comment (jedel1043) [inline boa_engine/src/value/conversions.rs:65]: Why isn't this mirroring the `f64` conversion?

Comment (raskad) [reply boa_engine/src/value/conversions.rs:65]: It is the same as the `f64` one no?

Comment (jedel1043) [reply boa_engine/src/value/conversions.rs:65]: Ah, right, forgot we removed the `f64` conversion to integer

**Review by jedel1043 — APPROVED** [review]
Thank you!

**Review by raskad — APPROVED** [review]

**Pull request closed without merging**